polyglot/piranha/src/cleanup_rules/go/rules.toml

```diff
@@ -0,0 +1,547 @@
+#   true != true   -> false
+#   false != false -> false
+#   nil != nil     -> false
+#
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_identity_not_equal"
+query = """
+(
+    (binary_expression
+        left: (
+            [
+                (false)
+                (parenthesized_expression (false))
+                (true)
+                (parenthesized_expression (true))
+                (nil)
+                (parenthesized_expression (nil))
+            ]
+        ) @lhs
+        operator: "!="
+        right: (
+            [
+                (false)
+                (parenthesized_expression (false))
+                (true)
+                (parenthesized_expression (true))
+                (nil)
+                (parenthesized_expression (nil))
+            ]
+        ) @rhs
+    ) @binary_expression
+    (#eq? @lhs @rhs)
+)
+"""
+replace = "false"
+replace_node = "binary_expression"
+
+# Dummy rule that acts as a junction for all statement based cleanups
+[[rules]]
+name = "statement_cleanup"
+
+# Before :
+#  if (true) { doSomething(); }
+# After :
+#  { doSomething(); }
+#
+# Before :
+#  if (true) { doSomething(); } else { doSomethingElse();}
+# After :
+#  { doSomething(); }
+#
+[[rules]]
+groups = ["if_cleanup"]
+name = "simplify_if_statement_true"
+query = """
+(
+    (if_statement
+        condition : (
+            [
+                (true)
+                (parenthesized_expression (true))
+            ]
+        )
+        consequence : ((block) @consequence)
+    ) @if_statement
+)
+"""
+replace = "@consequence"
+replace_node = "if_statement"
+
+# Before :
+#  if (true) { doSomething(); } else { doSomethingElse();}
+# After :
+#  { doSomethingElse(); }
+#
+# Before :
+#  if (true) { doSomething(); }
+# After :
+#
+[[rules]]
+groups = ["if_cleanup"]
+name = "simplify_if_statement_false"
+query = """
+(
+    (if_statement
+        condition : (
+            [
+                (false)
+                (parenthesized_expression (false))
+            ]
+        )
+        consequence : (_)
+        alternative: ((_) @alternative) ?
+    ) @if_statement
+)
+"""
+replace = "@alternative"
+replace_node = "if_statement"
+
+# Before :
+#  {
+#     someStepsBefore();
+#     {
+#        someSteps();
+#     }
+#     someStepsAfter();
+#  }
+# After :
+#  {
+#     someStepsBefore();
+#        someSteps();
+#     someStepsAfter();
+#  }
+#
+# Note that we need to tag basically all nodes here.
+# Including not so obvious ones: @outer.stmt_list and @outer.block
+[[rules]]
+name = "remove_unnecessary_nested_block"
+query = """
+(
+    (block
+        (statement_list
+            (_)* @pre
+            (block
+                (statement_list) @nested.statements
+            ) @nested.block
+            (_)* @post
+        ) @outer.stmt_list
+    ) @outer.block
+)
+"""
+replace = "@nested.statements"
+replace_node = "nested.block"
+
+# This rule is not deleting multiple statements after return.
```

Comment:
IS that some issue with the grammar? Can we look into it ?

Reply:
We can now successfully match all statements after `return_statement` in a similar way as e.g., Java and Javascript.
Accounting for the `\n`terminator did the trick: 

```
(((statement) "\\n"?)+ @post)
```

However, **replace** is only working for the first statement. This is also happening in Java (draft #X).
The self-edge makes it work for the current implementation.

Reply:
Introduced dummy node `return_cleanup` and now we have a cycle